crates/viewer/re_data_ui/src/instance_path.rs

```diff
@@ -278,16 +278,17 @@ fn preview_if_image_ui(
         buffer: components::ImageBuffer,
         format: components::ImageFormat
     );
-    static_assert_struct_has_fields!(
-        archetypes::DepthImage,
-        buffer: components::ImageBuffer,
-        format: components::ImageFormat
-    );
-    static_assert_struct_has_fields!(
-        archetypes::SegmentationImage,
-        buffer: components::ImageBuffer,
-        format: components::ImageFormat
-    );
+    // TODO(#7245): How to check this with eager serialized archetypes?
+    // static_assert_struct_has_fields!(
```

Comment:
We can still check that is has the fields, but not their types.

However, you can replace it with a runtime-check instead (in debug builds), using the reflection API

Reply:
did the real thing now :)
`debug_assert_archetype_has_components`